src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php

```diff
@@ -70,19 +86,230 @@ public function indexAction(Request $request)
             'help_link' => $this->generateSidebarLink($request->get('_legacy_controller')),
             'requireFilterStatus' => false,
             'form' => $form->createView(),
+            'grid' => $presentedGrid,
         ];
 
         return $this->render('@AdvancedParameters/WebservicePage/webservice.html.twig', $twigValues);
     }
 
+    /**
+     * Redirects to webservice account form where new webservice account record can be created.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller')~'_')")
```

Comment:
you don't need to add `~'_'` anymore, can you remove them to all your uses of AdminSecurity annotation please?

Reply:
could we go one step further and make `request.get('_legacy_controller')` optional? by default it could use `_legacy_controller` for access control check?